src/bun.js/bindings/JSBuffer.cpp

```diff
@@ -2400,8 +2414,14 @@ JSC_DEFINE_HOST_FUNCTION(constructJSBuffer, (JSC::JSGlobalObject * lexicalGlobal
 
             RELEASE_AND_RETURN(throwScope, JSC::JSValue::encode(uint8Array));
         }
-        default: {
+        case ArrayType:
```

Comment:
Is that the only type we need to worry about here? What about DerivedArrayType? There are likely 10 more

Reply:
good catch, will add DerivedArrayType

<img width="439" alt="image" src="https://github.com/user-attachments/assets/6a1cc2fe-0539-40ab-ace2-22094dc9746a">

Reply:
there are more, it should really be a "if we don't have any other things to match against" instead of assuming